### PR TITLE
Consolidate bug_report_templates and remove the gem versions

### PR DESCRIPTION
**Pull request by zzak**
As a contributor, changing a single line to test the desired version does not warrant a separate file for each type of report.

The only argument I can see for keeping the gem versions is that users are likely reporting a bug for a specific version of Rails, typically during an upgrade. As few people are probably testing their applications against Rails main branch.

Additionally, having the gem version templates means that we are testing both main and a stable release of Rails in CI, which has some benefit.

IMO, the cost of separate files and the confusion it will create if we want to add more templates outweighs the benefits mentioned here.

My motivation is that I want to add more report templates, like for additional adapters, and this list will grow to be unmanageable if we keep the gem versions.

/cc @yahonda @skipkayhil as I think we've discussed this before

### CODE REVIEW

**Review by skipkayhil — APPROVED** [review]
:+1: easier maintenance and I usually end up starting with one and adding the other anyways

**Comment by yahonda** [comment]
I think these bug report templates are not only for active contributors, sometimes they are first time contributors.

For the most of first time contributors, I wanted bug report templates can be used without updating `gem "rails"` line.

Changing the Rails version `gem "rails", "~> 7.1.0"` every time the newer version of Rails released should not be so hard.

**Comment by byroot** [comment]
Makes sense to me. But yeah, maybe the default should be the last major, and edge be the one commented out?

We wouldn't even need to update the version in the template, we can just leave it out: `gem "rails"`

**Comment by JohnNewbold** [comment]
First time

**Comment by zzak** [comment]
Thanks for your feedback! 😄 
I've set them to just use `gem "rails"` with a comment about testing against main.

The only downside to this right now is we lose coverage of CI running main against these bug reports, but that is a problem I'd like to fix in `buildkite-config` once that refactor is merged.

File: guides/bug_report_templates/action_controller.rb [hunk 1]
```diff
@@ -7,17 +7,16 @@
...
+  # gem "rails", github: "rails/rails", branch: "main"
 end
 
-require "rack/test"
 require "action_controller/railtie"
 
 class TestApp < Rails::Application
   config.root = __dir__
   config.hosts << "example.org"
-  config.session_store :cookie_store, key: "cookie_store_key"
```
Comment (byroot) [inline guides/bug_report_templates/action_controller.rb:20]: Unrelated but should we add `config.load_defaults`?

Comment (skipkayhil) [reply guides/bug_report_templates/action_controller.rb:20]: :+1: A bunch of others templates had it added in #49631

**Comment by JohnNewbold** [comment]
Thanks

**Comment by ghiculescu** [comment]
@zzak since this change, the links on https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#create-an-executable-test-case don't work.

**Comment by zzak** [comment]
@ghiculescu Good catch, and thanks for the ping. Backported the guide changes to 7-1-stable in #50550 and 7-0-stable in #50551 :pray:

**Pull request merged**